src/Microsoft.AspNetCore.Http.Abstractions/HostString.cs

```diff
@@ -208,6 +205,66 @@ public static HostString FromUriComponent(Uri uri)
                 UriComponents.HostAndPort, UriFormat.Unescaped));
         }
 
+        /// <summary>
+        /// Matches the host portion of a host header value against a list of patterns.
+        /// The host may be the encoded punycode or decoded unicode form so long as the pattern
+        /// uses the same format.
+        /// </summary>
+        /// <param name="value">Host header value with or without a port.</param>
+        /// <param name="patterns">A set of pattern to match, without ports.</param>
+        /// <remarks>
+        /// The port on the given value is ignored. The patterns should not have ports.
```

Comment:
Would it be better to reject patterns with ports to make this more clear? A downside is that whatever is translates binding config into to patterns would have to strip the ports.

Reply:
I misunderstood. Failing to ever match is OK.